### ISSUE TITLE
xvi hasn't been valground

### ISSUE THREAD

**Issue by martinwguy** [martinwguy/xvi#59]
Valgrind, just pressing : says
```
==6511== Conditional jump or move depends on uninitialised value(s)
==6511==    at 0x8050E9B: normal (in /home/martin/xvi/src/xvi)
==6511==    by 0x804D78C: n_proc (in /home/martin/xvi/src/xvi)
==6511==    by 0x804DA1D: xvi_handle_event (in /home/martin/xvi/src/xvi)
==6511==    by 0x805AF01: tcap_scr_main (in /home/martin/xvi/src/xvi)
==6511==    by 0x80490EA: main (in /home/martin/xvi/src/xvi)
```
It would be worth investigating this and anything else it might turn up.

**Comment by jvvv** [comment]
I must be using different options. I get:
```
==28373== Conditional jump or move depends on uninitialised value(s)
==28373==    at 0x40A671: flexaddch (flexbuf.c:39)
==28373==    by 0x40A72F: flexgetstr (flexbuf.c:70)
==28373==    by 0x4124F4: do_sline (screen.c:288)
==28373==    by 0x412479: update_sline (screen.c:263)
==28373==    by 0x4156D3: show_file_info (status.c:125)
==28373==    by 0x4156FF: info_update (status.c:132)
==28373==    by 0x40D6C3: move_cursor (movement.c:315)
==28373==    by 0x4150FA: xvi_startup (startup.c:312)
==28373==    by 0x41D7C9: tcap_scr_main (tcap_scr.c:282)
==28373==    by 0x41D6F0: main (tcapmain.c:31)
==28373==  Uninitialised value was created by a heap allocation
==28373==    at 0x4C28B5F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==28373==    by 0x401BF1: alloc (alloc.c:147)
==28373==    by 0x41B790: add_window (windows.c:695)
==28373==    by 0x41A98E: xvInitWindow (windows.c:41)
==28373==    by 0x414BFB: xvi_startup (startup.c:104)
==28373==    by 0x41D7C9: tcap_scr_main (tcap_scr.c:282)
==28373==    by 0x41D6F0: main (tcapmain.c:31)
```
and
```
==28373== Conditional jump or move depends on uninitialised value(s)
==28373==    at 0x40A671: flexaddch (flexbuf.c:39)
==28373==    by 0x40A72F: flexgetstr (flexbuf.c:70)
==28373==    by 0x409113: get_file (fileio.c:486)
==28373==    by 0x406AEF: exEditFile (ex_cmds1.c:431)
==28373==    by 0x407163: exNext (ex_cmds1.c:661)
==28373==    by 0x415166: xvi_startup (startup.c:329)
==28373==    by 0x41D7C9: tcap_scr_main (tcap_scr.c:282)
==28373==    by 0x41D6F0: main (tcapmain.c:31)
==28373==  Uninitialised value was created by a stack allocation
==28373==    at 0x408BFB: get_file (fileio.c:245)
```
and
```
==28373== Conditional jump or move depends on uninitialised value(s)
==28373==    at 0x40DD68: normal (normal.c:103)
==28373==    by 0x40886B: n_proc (events.c:244)
==28373==    by 0x40871D: xvi_handle_event (events.c:179)
==28373==    by 0x41D95E: tcap_scr_main (tcap_scr.c:355)
==28373==    by 0x41D6F0: main (tcapmain.c:31)
==28373==  Uninitialised value was created by a heap allocation
==28373==    at 0x4C28B5F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==28373==    by 0x401BF1: alloc (alloc.c:147)
==28373==    by 0x41B6F6: alloc_window (windows.c:656)
==28373==    by 0x41A9CB: xvInitWindow (windows.c:49)
==28373==    by 0x414BFB: xvi_startup (startup.c:104)
==28373==    by 0x41D7C9: tcap_scr_main (tcap_scr.c:282)
==28373==    by 0x41D6F0: main (tcapmain.c:31)
```

I'm using these options with valgrind:
```
 valgrind --tool=memcheck -v --log-file=valgrind.log --read-var-info=yes --read-inline-info=yes --leak-check=no --track-origins=yes --show-mismatched-frees=yes ./src/xvi
```

The fixes for these are trivial, but worth adding. I can push a branch now, or I can wait to see if more like these show up.

**Comment by martinwguy** [comment]
Yes, please do so. valgrind should be silent, so that any new similar
bugs are evident, sort of like silencing compiler warnings.
If it gets down to 0 errors, I can add it to the testsuite and we get
automatic testing for the things valgrind notices.

**Comment by martinwguy** [comment]
Fixed by 40fec7cf
